File: csharp/ToolGood.Algorithm/Internals/Functions/String/Function_CHAR.cs
using System;
using System.Collections.Generic;
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm.Internals.Functions.String
{
	internal sealed class Function_CHAR : Function_1
	{
		public Function_CHAR(FunctionBase[] funcs) : base(funcs)
		{
			if (funcs.Length != 1) {
				throw new ArgumentException($"Function '{Name}' requires exactly 1 parameter.");
			}
		}

		public override string Name => "Char";

		public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
		{
			var args1 = GetNumber_1(engine, tempParameter);
			if (args1.IsErrorOrNone) { return args1; }
			var code = args1.IntValue;
			if (code < 0 || code > 65535) {
				return ParameterError(1);
			}
			char c = (char)code;
			return Operand.Create(new string(c, 1));
		}
		public override OperandType GetResultType()
		{
			return OperandType.TEXT;
		}

		internal override void GetParameterTypes(NoneEngine noneEngine, List<ParameterType> result, OperandType operandType, string op = null, string val = null)
		{
			func1.GetParameterTypes(noneEngine, result, OperandType.NUMBER);
		}
	}

}

File: JavaScript/src/Internals/Functions/String/Function_CHAR.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';

class Function_CHAR extends Function_1 {
    get Name() {
        return "Char";
    }

    constructor(a) {
        super(a);
    }

    evaluate(work, tempParameter) {
        let args1 = this.getNumber_1(work, tempParameter);
        if (args1.IsError) { return args1; }
        let code = args1.IntValue;
        // 与 C# 一致:超出 0~65535 时参数错误(JS fromCharCode 越界会静默取模)
        if (code < 0 || code > 65535) {
            return this.parameterError(1);
        }
        let c = String.fromCharCode(code);
        return Operand.Create(c);
    }
}

export { Function_CHAR };

